File: commandapi-documentation-velocity-code/src/main/java/dev/jorel/commandapi/examples/java/Examples.java
package dev.jorel.commandapi.examples.java;

import com.google.inject.Inject;
import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent;
import com.velocitypowered.api.proxy.ProxyServer;
import dev.jorel.commandapi.CommandAPI;
import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.CommandAPIVelocityConfig;
import dev.jorel.commandapi.arguments.IntegerArgument;
import net.kyori.adventure.text.Component;
import org.slf4j.Logger;

import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;

@SuppressWarnings("InnerClassMayBeStatic")
public class Examples {

class velocityIntro {
public class ExamplePlugin {

    private final ProxyServer server;
    private final Logger logger;

/* ANCHOR: velocityIntro1 */
@Inject
public ExamplePlugin(ProxyServer server, Logger logger) {
    this.server = server;
    this.logger = logger;

    CommandAPI.onLoad(new CommandAPIVelocityConfig(server, this));
}
/* ANCHOR_END: velocityIntro1 */

/* ANCHOR: velocityIntro2 */
@Subscribe
public void onProxyInitialization(ProxyInitializeEvent event) {
    // Any CommandAPI command registrations...
    CommandAPI.onEnable();
}
/* ANCHOR_END: velocityIntro2 */
}

void velocityIntro() {
/* ANCHOR: velocityIntro3 */
new CommandAPICommand("randomnumber")
    .withArguments(new IntegerArgument("min"))
    .withArguments(new IntegerArgument("max"))
    .executesPlayer((player, args) -> {
        int min = (int) args.get("min");
        int max = (int) args.get("max");
        Random random = ThreadLocalRandom.current();
        int randomNumber = random.nextInt(min, max);
        player.sendMessage(Component.text().content("Your random number is: " + randomNumber));
    })
    .register();
/* ANCHOR_END: velocityIntro3 */
}
}
}

File: commandapi-documentation-velocity-code/src/main/kotlin/dev/jorel/commandapi/examples/kotlin/Examples.kt
package dev.jorel.commandapi.examples.kotlin

import com.google.inject.Inject
import com.velocitypowered.api.event.Subscribe
import com.velocitypowered.api.event.proxy.ProxyInitializeEvent
import com.velocitypowered.api.proxy.ProxyServer
import dev.jorel.commandapi.CommandAPI
import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.CommandAPIVelocity
import dev.jorel.commandapi.CommandAPIVelocityConfig
import dev.jorel.commandapi.arguments.IntegerArgument
import dev.jorel.commandapi.executors.NormalExecutor
import net.kyori.adventure.text.Component
import org.slf4j.Logger
import java.util.concurrent.ThreadLocalRandom
import kotlin.random.Random

class Examples {

fun velocityIntro() {
/* ANCHOR: velocityIntro1 */
CommandAPICommand("randomnumber")
    .withArguments(IntegerArgument("min"))
    .withArguments(IntegerArgument("max"))
    .executesPlayer(NormalExecutor { player, args ->
        val min = args["min"] as Int
        val max = args["max"] as Int
        val random = ThreadLocalRandom.current()
        val randomNumber = random.nextInt(min, max)
        player.sendMessage(Component.text().content("Your random number is: $randomNumber"))
    })
    .register()
/* ANCHOR_END: velocityIntro1 */
}

}